src/Layout/Page/Page.component.tsx

```diff
@@ -2,50 +2,215 @@
 import * as React from 'react';
 // VENDOR
 import styled from 'styled-components';
-// import { Box } from '@rebass/grid';
-// STYLES
-import { colors, sizes } from '../../theme';
+import { Grid, Cell } from 'styled-css-grid';
+import classnames from 'classnames';
 // COMPONENTS
-import Header from '../Header/Header.component';
-import Footer from '../Footer/Footer.component';
 import { NormalizeCSS, GlobalCSS } from '../../theme/GlobalStyles';
+import { CustomAdaptor, StandardBreakpoints } from '../../Grid/Adaptor';
 
-const StyledPageElement = styled.section`
+// Constants
+// ------------------------------------------------------------------------------------------------------------------
+export const DEFAULT_LAYOUT_WIDTH = '100%';
+export const DEFAULT_CONTENT_WIDTH = '71.25rem';
+export const DEFAULT_SIDEBAR_WIDTH = '13.75rem';
+export const TRANSPARENT = 'transparent';
+export const DEFAULT_HEADER_HEIGHT = '1.5rem';
+export const DEFAULT_FOOTER_HEIGHT = '20.375rem';
+const RIGHT = 'right';
+
+// Defaults
+// ------------------------------------------------------------------------------------------------------------------
+const PageDefaultProps = {
+    headerHeight: DEFAULT_HEADER_HEIGHT,
+    footerHeight: DEFAULT_FOOTER_HEIGHT,
+};
+
+const DefaultLayoutDefaultProps = {
+    layoutWidth: DEFAULT_LAYOUT_WIDTH,
+    contentWidth: DEFAULT_CONTENT_WIDTH,
+    layoutBackgroundColor: TRANSPARENT,
+    contentBackgroundColor: TRANSPARENT,
+};
+
+const SidebarLayoutDefaultProps = {
+    sidebarWidth: DEFAULT_SIDEBAR_WIDTH,
+    layoutWidth: DEFAULT_CONTENT_WIDTH,
+    contentWidth: DEFAULT_CONTENT_WIDTH,
+    layoutBackgroundColor: TRANSPARENT,
+    contentBackgroundColor: TRANSPARENT,
+};
+
+// Types
+// ------------------------------------------------------------------------------------------------------------------
+type SidebarAlignment = 'left' | 'right';
+
+// Styled Components
+// ------------------------------------------------------------------------------------------------------------------
+const StyledPage = styled.section`
     display: flex;
     flex-direction: column;
     height: 100vh;
-    max-width: 100vw;
-
-    header {
-        flex: 0 0 auto;
-        max-width: 100vw;
-    }
+    max-width: 100%;
+`;
 
-    footer {
-        flex: 0 0 auto;
-        max-width: 100vw;
-    }
+const StyledLayout = styled.section<StyledLayoutProps>`
+    max-width: ${props => props.layoutWidth};
+    background-color: ${props => props.layoutBackgroundColor};
+    margin: 0 auto;
 `;
 
-const StyledBodyElement = styled.section`
-    flex: 1 0 auto;
-    background-color: ${colors.silver.base};
-    max-width: 100vw;
-    align-self: stretch;
-    padding: ${sizes.padding.lg} ${sizes.padding.xl};
+const StyledContent = styled.div<StyledContentProps>`
```

Comment:
We may wanna use a `main` here in place of a `div` for SEO and best practices: https://www.w3schools.com/tags/tag_main.asp

Reply:
I think this is a good idea. I wasn't crazy about using `div` or `section` for the main wrapper. Ironically the `main` tag was looking me right in the face and I didn't even think of it.

Reply:
Yeah, I often lose sight of the trees in the forrest. I'm pretty sure the content area for Rx is inside of a `section` 🙈